src/main/java/com/yahoo/memory/UnsafeUtil.java

```diff
@@ -13,26 +13,30 @@
 /**
  * Provides access to the sun.misc.Unsafe class and its key static fields.
  *
- * <p>The internal static initializer also detects whether the methods unique to the Unsafe class in
- * JDK8 are present; if not, methods that are compatible with JDK7 are substituted using an internal
- * interface.  In order for this to work, this library still needs to be compiled using jdk8
- * and it must be done with both source and target versions of jdk7 specified in pom.xml.
+ * <p>The internal static initializer also detects whether the methods unique to the Unsafe class
+ * in JDK8 are present; if not, methods that are compatible with JDK7 are substituted using an
+ * internal interface.  In order for this to work with jdk7, this library must be compiled using
+ * jdk8 and it must be done with both source and target versions of jdk7 specified in pom.xml.
  * The resultant jar will work on jdk7 and jdk8.</p>
  *
+ * <p>This may work with jdk9 but might require the JVM arg <i>-permit-illegal-access</i>,
+ * <i>–illegal-access=permit</i> or equivalent. Proper operation with jdk9 or above is not
+ * guaranteed and has not been tested.
+ *
  * @author Lee Rhodes
  */
 public final class UnsafeUtil {
   public static final Unsafe unsafe;
-  public static final int JDK;
+  public static final double JDK;
```

Comment:
Could we store this as String? Double is not precise and dangerous.

Reply:
Good point.  I wanted it numeric and monotonically comparable. I will fix.

Reply:
If you want versions to be numeric just to be able to compare them, note that you can compare strings as well like `version.compareTo("1.8") > 0`.

Reply:
There is a problem attempting to simply convert these version numbers to strictly numeric numbers.  E.g., a version 1.10 logically greater than 1.9 but is numerically less.  

I have a more robust fix, but haven't checked it in yet because of these 6 other failures.